### PR TITLE
[dv/top] Adjust test parameters

### PR DESCRIPTION
**Pull request by tjaychen**
The pause test is currently failing because the test_rom takes
too long to execute and the watchdog triggers sooner than the
test expects. As a result, the test sees two reset reasons at
the same time and fails.

While we could adjust the delays such that it is passing again,
such tests may be prone to failures in the future.

Instead, the test was converted to not rely on resets but instead
on the watchdog / wakeup counts.  The pause function causes the
watchdog to stop counting during low power, and thus creates an
inverse relationship with wakeup.  The test checks on that instead.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
let me know if this change seems reasonable to you guys.

**Comment by engdoreis** [comment]
Thanks @tjaychen for the PR.
I prefer to have the test checking that a reset has not occurred, as it also checks the integration between the IPs.
My suggestion would be changing the test to use the `Normal sleep` rather than the `deep_sleep`. In the first, the software continues executing from the last instruction, thus the `test_rom` would not take time during the wake up.
https://docs.opentitan.org/hw/ip/pwrmgr/doc/index.html#programmers-guide

**Comment by tjaychen** [comment]
> `eep_sleep`. In the first, the software continues executing from the last instruction, thus the `test_rom` boot time would not interfe

i actually thought about that, and was originally going down that path. But do you see anything that is not tested with this approach that is tested with reset reasons not seen?  To me looking at the count is sufficient, since consistency between count and the behavior of aon_timer reset request should already be tested at block level. 

In general though, all these tests need to be tested with both deep sleep and normal sleep, see #12816

**Review by rswarbrick — APPROVED** [review]
This looks reasonable to me, but maybe it's worth checking with @matutem since he wrote it in the first place? (That said, I have a vague recollection that Matute is out at the moment: if so, no worries)

**Comment by tjaychen** [comment]
wait i actually thought @doreis-lowrisc wrote it.
But let's check with @matutem also.

File: sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c [hunk 1]
```diff
@@ -50,21 +49,28 @@ bool test_main(void) {
...
+
+    // The actual expiration of the watchdog is unimportant, as the test
+    // mainly checks the count
+    aon_timer_testutils_watchdog_config(&aon_timer, UINT32_MAX, UINT32_MAX,
                                         true);
+    aon_timer_testutils_wakeup_config(&aon_timer, wkup_cycles);
+
+    busy_spin_micros(IDLE_TIME_US);
+
+    // Since watchdog was started first, its count should be larger
```
Comment (engdoreis) [inline sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c:69]: Same here.

File: sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c [hunk 2]
```diff
@@ -74,24 +80,13 @@ bool test_main(void) {
...
-    CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
-
-    CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(
-        &aon_timer, kDifAonTimerIrqWkupTimerExpired));
-    busy_spin_micros(WDOG_BITE_TIME_US - WDOG_BARK_TIME_US + 10);
-    LOG_ERROR("Should have got a watchdog reset");
-  } else if (rst_info == kDifRstmgrResetInfoWatchdog) {
-    LOG_INFO("Booting for the third time due to wdog bite reset");
+    // Since watchdog count was paused during low power, wakeup
+    // count should now be larger
```
Comment (engdoreis) [inline sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c:85]: Same here

**Review by engdoreis — APPROVED** [review]

**Review by timothytrippel — APPROVED** [review]

**Review by matutem — APPROVED** [review]
Left a question, otherwise LGTM

File: sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c [hunk 3]
```diff
@@ -50,21 +49,28 @@ bool test_main(void) {
...
         aon_timer_testutils_get_aon_cycles_from_us(WKUP_TIME_US);
-    uint32_t bark_cycles =
-        aon_timer_testutils_get_aon_cycles_from_us(WDOG_BARK_TIME_US);
-    uint32_t bite_cycles =
-        aon_timer_testutils_get_aon_cycles_from_us(WDOG_BITE_TIME_US);
-    aon_timer_testutils_wakeup_config(&aon_timer, wkup_cycles);
-    aon_timer_testutils_watchdog_config(&aon_timer, bark_cycles, bite_cycles,
+
+    // The actual expiration of the watchdog is unimportant, as the test
+    // mainly checks the count
```
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/device/tests/aon_timer_sleep_wdog_sleep_pause_test.c:62]: I mean finish the comment with a `.`

**Comment by tjaychen** [comment]
thanks @doreis-lowrisc , @rswarbrick @matutem for the review.
If you guys think this test diminishes the original intent, just let me know, i can update as needed.

**Pull request merged**